parcels/include/index_search.h

```diff
@@ -260,6 +300,13 @@ static inline StatusCode search_indices_rectilinear(type_coord x, type_coord y,
   else
     *zeta = 0;
 
+  if ( (*xsi < 0)  && (is_zero_dbl(*xsi)) )       {*xsi = 0.;}
+  if ( (*xsi > 1)  && (is_close_dbl(*xsi, 1.)) )  {*xsi = 1.;}
+  if ( (*eta < 0)  && (is_zero_dbl(*eta)) )       {*eta = 0.;}
+  if ( (*eta > 1)  && (is_close_dbl(*eta, 1.)) )  {*eta = 1.;}
+  if ( (*zeta < 0) && (is_zero_dbl(*zeta)) )      {*zeta = 0.;}
+  if ( (*zeta > 1) && (is_close_dbl(*zeta, 1.)) ) {*zeta = 1.;}
```

Comment:
That's indeed quite good - indeed should prevent some barycentric interpolation errors close the the cells' edges.